### PR TITLE
Add support for multitype mocks

### PR DESCRIPTION
**Pull request by enumag**
Currently PHPStan understands

```php
$handle = Phony::mock(InterfaceA::class)->get();
```

but doesn't understand

```php
$handle = Phony::mock([InterfaceA::class, InterfaceB::class, TraitA::class])->get();
```

I'm trying to fix it in this PR.

### CODE REVIEW

File: src/Type/MockReturnType.php [hunk 1]
```diff
@@ -74,27 +75,48 @@ private function getTypeFromCall(
...
+
+        if (!$classes) {
             return $reflection->getReturnType();
         }
 
-        $class = $arg->class;
+        return new InstanceHandleType(...$classes);
+    }
+
+    private function getTypeFromClassConst(ClassConstFetch $classConstFetch, Scope $scope): ?string
```
Comment (ezzatron) [inline src/Type/MockReturnType.php:101]: This doesn't really return a "type" in terms of PHPStan terminology. A better name might be `getNameFromClassConst` or `getClassNameFromConst`.

Also, this repo currently supports PHP 7.0, so nullable types can't be used unfortunately (this is why the Travis build is failing btw).

**Review by ezzatron — CHANGES_REQUESTED** [review]
Thanks for the PR! Definitely interested in getting these improvements merged.

There are a couple of things I'd like to address before merging:

1. These changes really need tests. You can add tests by dropping new fixtures in the `test/fixture/type` directory. I will describe how the tests work in another comment, but basically you can copy an existing test and go from there.
2. The same change should also be applied to the `MockBuilderReturnType` class. It has similar logic for determining types. It might be best to refactor the logic into a trait, to keep it DRY going forward.

**Comment by ezzatron** [comment]
Each test fixture is a directory, where the name of the directory gives a reasonable indication of what the test intends to demonstrate. There are 4 files in each test fixture:

- `description` is a plaintext file, with a human-readable description of the test.
- `source.php` is the source code to be inspected by PHPStan.
- `with` is the expected output **with** the `phpstan-phony` integration.
- `without` is the expected output **without** the `phpstan-phony` integration.

Depending on what's being tested, the `with` or `without` files may need to be empty. For example, if you're testing that `phpstan-phony` prevents some errors from occurring, the `with` file will probably be empty. Whereas if you're testing that `phpstan-phony` raises additional errors, the `without` file will probably be empty instead.

**Comment by enumag** [comment]
Thanks for detailed comments. I'll try to address those issues.

**Comment by ezzatron** [comment]
While we're doing PRs, any chance you feel like progressing this one?

**Comment by enumag** [comment]
Not right now. I want to upgrade to PHPStan 0.10 first.

**Comment by ezzatron** [comment]
Might as well bump this PR while I'm here 😉

**Comment by ezzatron** [comment]
Multi-type mocks are now supported in [0.6.0](https://github.com/eloquent/phpstan-phony/releases/0.6.0).

**Pull request closed without merging**